### ISSUE TITLE
Scheduling early transition to Workbench

### ISSUE THREAD

**Issue by tobyhodges** [LibraryCarpentry/lc-spreadsheets#134]
Short version: **are you are willing to volunteer your lesson for early transition to The Carpentries Workbench 
infrastructure?**

---

Longer version below:

As I hope you are already aware, @zkamvar and the rest of the Curriculum Team are preparing to roll out the new lesson infrastructure, [The Carpentries Workbench](https://carpentries.github.io/workbench/), across all of The Carpentries official lessons in early May 2023. This means that all Data Carpentry, Library Carpentry, and Software Carpentry lesson repositories will be modified to adopt the new infrastructure at the end of this month.

As you might imagine, coordinating a rollout like this involves a large amount of time and effort, and one thing that will really help us to keep to the schedule and avoid disruption for the community is to be able to **prepare some lessons for 
transition before the 1st May deadline**.

With this in mind, **I invite you to volunteer to schedule an early transition for theis lesson repository.** If you expect to have time to adopt the Workbench version of the lesson repository in the next couple of weeks, please reply to let me know. As lesson Maintainers, these are the things you will need to be prepared to do for the transition:
1. **Be prepared to quickly merge pull requests from Zhian**. These pull requests will help update the lesson and aid the transition process.
2. **Close any outstanding pull requests.** Your repository currently has very few or zero open PRs, but we cannot rule out the possibility that someone will open more in the time between now and when the transition takes place. [Open PRs will be invalidated when the transition takes place](https://carpentries.github.io/sandpaper-docs/pull-request.html#transition-from-carpentriesstyles).
3. **Preview the lesson (in the https://github.com/fishtree-attempt organisation) when it is available**. Zhian will open an issue in your repository, tagging you when the preview is available. You should reply on that issue if you notice any problems with the Workbench version of the lesson site and repository in the preview.
4. Just before the transition happens, the repository will be temporarily set as read-only and an issue will be opened (see an example from the Maintainer Onboarding curriculum: https://github.com/carpentries/maintainer-onboarding/issues/69) 
5. After the transition, **delete and recreate any forks and local clones** you have of the repository and then **confirm that you have done so** (you will be given specific instructions in an issue).

If you are willing and able to help us ensure a smooth rollout of the Workbench, please reply to let me know that you are happy to volunteer your lesson for early transition. If there is a specific date you would like the transition to take place (or that we should avoid), please also mention that.

**Comment by morskyjezek** [comment]
Hi Toby, yes I'd be happy to support this transition and help with this lesson.

**Comment by tobyhodges** [comment]
Thank you @morskyjezek 🙌 

> I'd also welcome the chance to learn more about the new lesson framework.

@zkamvar will host two community discussions on the Workbench on Monday next week (16:00 & 22:00 UTC) - see the etherpad for more details. I will also lead a "the Workbench from a Maintainer's perspective" section in Maintainer Onboarding later this month. Happy to share details of that session if you would like to join.

For a more asynchronous option, I think [Zhian's transition guide](https://carpentries.github.io/workbench/transition-guide.html) provides a nice overview of the main things that are changing for Maintainers.

**Comment by morskyjezek** [comment]
I think I can make the later session next week and we'll have a look at the guide. Thanks!

**Comment by tobyhodges** [comment]
Hi @morskyjezek we are getting ready for an early transition for this lesson, and we have prepared [a schedule for early lesson transitions](https://docs.google.com/spreadsheets/d/1Y_ZdAZu11Lc1bcInE9801ax6khvgFzjTh70P3v8crL8/edit?usp=sharing). 

Please check that document to see the date we have planned for transitioning this lesson, and adjust the sheet if a different date would be more convenient for you (no more than three lessons per day, please!). The README sheet gives more information. If we don't hear from you, we will go ahead with the transition on the date shown in that document.

**Comment by morskyjezek** [comment]
Yes, I think that 4/21 looks okay for this lesson. I'm also reviewing th README and guide

**Comment by zkamvar** [comment]
Hello, This is just to note that I will transition this today. The tracking issue is in https://github.com/carpentries/lesson-transition/issues/68

In the next few hours, I will make two comments (the first with instructions) before and after the transition, respectively.

**Comment by zkamvar** [comment]
This lesson will be converted to use [The Carpentries Workbench][workbench]
To prevent accidental reversion of the changes, we are temporarily revoking
write access for all collaborators on this lesson:

 - [ ] @jt14den (admin)
 - [ ] @jezcope (admin)
 - [ ] @ostephens (admin)
 - [ ] @cmacdonell (admin)
 - [ ] @weaverbel (admin)
 - [ ] @shlake (push)
 - [ ] @drjwbaker (admin)
 - [ ] @morskyjezek (push)
 - [ ] @ccronje (admin)
 - [ ] @U2NG (admin)
 - [ ] @pitviper6 (admin)

If you no longer wish to have write access to this repository, you do not
need to do anything further.

1. What you can expect from the transition 📹: https://carpentries.github.io/workbench/beta-phase.html#beta
2. How to update your local clone 💻: https://carpentries.github.io/workbench/beta-phase.html#updating-clone
3. How to update (delete) your fork (if you have one) 📹: https://carpentries.github.io/workbench/faq.html#update-fork-from-styles

If you wish to regain write access, please re-clone the repository on your machine and
then comment here with `I am ready for write access :rocket:` and the
admin maintainer of this repository will restore your permissions.

If you have any questions, please reply here and tag @zkamvar

[workbench]: https://carpentries.github.io/workbench

**Comment by zkamvar** [comment]
The deed is done. The infrastructure takes a few minutes to bootstrap and cache the packages for the lesson build. Once the build is done, I will switch github pages to deploy from the `gh-pages` branch and you will have your workbench lesson. 

Thank you all for your enthusiasm and your patience!

**Comment by zkamvar** [comment]
The Workbench version is now live: https://librarycarpentry.github.io/lc-spreadsheets/

In addition, here is [map of commits that were changed during the transition](https://github.com/carpentries/lesson-transition/blob/release_lc/lc-spreadsheets/release/librarycarpentry/lc-spreadsheets-commit-map.hash)

**Comment by zkamvar** [comment]
Hello @LibraryCarpentry/lc-spreadsheets-maintainers, I would just like to bump  https://github.com/LibraryCarpentry/lc-spreadsheets/issues/134#issuecomment-1518339136 as a reminder that you need to follow the post-transition instructions and comment here with `I am ready for write access :rocket:` to regain write permissions for this repository.

**Comment by morskyjezek** [comment]
Thanks for the reminder @zkamvar - I have updated and re-cloned a new repo on my local machine. 

I am ready for write access 🚀

**Comment by zkamvar** [comment]
Congratulations @morskyjezek, you now have write access 🎉

**Comment by ErinBecker** [comment]
Hi @jt14den - You have not yet responded here to report that you completed the steps listed by @zkamvar above to regain your Maintainer access to this repository. The lesson infrastructure transition required a second Maintainer team to be temporarily created for all lessons, which can only be removed when all Maintainers have confirmed that they are ready to have their access restored. Maintenance of both teams is not sustainable for The Carpentries in the long run. At the beginning of **November 2023** (note the deadline has been extended) we will remove access for Maintainers who have not taken the necessary action to continue working on the repository.

The Curriculum Team is truly grateful for everything you have already done as a Maintainer and we do not want to lose you! But we also understand that circumstances change and you may no longer have capacity to continue in your role. If you wish to retain your access and keep supporting the community, please follow the steps above then confirm that you are ready by commenting I am ready for write access 🚀 on this thread before the end of your day on **Wednesday 1st November.**

**Comment by ErinBecker** [comment]
Hi @jt14den  - You no longer have Maintainer access to this repository. Please see the comment above for details. If you would like to restore your access, please re-clone the repository on your machine and then comment here with I am ready for write access 🚀 and the admin maintainer of this repository will restore your permissions. If you have any questions, please don’t hesitate to get in touch [curriculum@carpentries.org](mailto:curriculum@carpentries.org)

To the remaining Maintainers for this lesson - the Curriculum Team will be prioritizing onboarding of new Maintainers for lessons that have lost Maintainers in our next round of onboarding. Please don’t hesitate to reach out if your lesson needs additional support in the meantime - [curriculum@carpentries.org](mailto:curriculum@carpentries.org).

I will close this issue now but will still be notified of any replies on the thread in future.
